packages/gitbook-v2/src/lib/data/lookup.ts

```diff
@@ -1,55 +1,115 @@
 import { race, tryCatch } from '@/lib/async';
 import { joinPath, joinPathWithBaseURL } from '@/lib/paths';
 import { trace } from '@/lib/tracing';
-import type { PublishedSiteContentLookup } from '@gitbook/api';
+import type { GitBookAPI, PublishedSiteContentLookup, SiteVisitorPayload } from '@gitbook/api';
 import { apiClient } from './api';
 import { getExposableError } from './errors';
 import type { DataFetcherResponse } from './types';
 import { getURLLookupAlternatives, stripURLSearch } from './urls';
 
+interface LookupPublishedContentByUrlInput {
+    url: string;
+    redirectOnError: boolean;
+    apiToken: string | null;
+}
+
+type FetchLookupAPIResponse =
+    | {
+          data?: undefined;
+          error: Error;
+      }
+    | {
+          data: Awaited<ReturnType<GitBookAPI['urls']['resolvePublishedContentByUrl']>>;
+          error?: undefined;
+      };
+
+/**
+ * Lookup a content by its URL using the GitBook resolvePublishedContentByUrl API endpoint.
+ * To optimize caching, we try multiple lookup alternatives and return the first one that matches.
+ */
+export async function resolvePublishedContentByUrl(
+    input: LookupPublishedContentByUrlInput & { visitorPayload: SiteVisitorPayload }
+) {
+    return lookupPublishedContentByUrl({
+        url: input.url,
+        fetchLookupAPIResult: ({ url, signal }) => {
+            const api = apiClient({ apiToken: input.apiToken });
+            return trace(
+                {
+                    operation: 'resolvePublishedContentByUrl',
+                    name: url,
+                },
+                () =>
+                    tryCatch(
+                        api.urls.resolvePublishedContentByUrl(
```

Comment:
Have we tested that the CF caching is working as expected on these POST?

Reply:
Yes I had ran some tests and the cache was busted properly